src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchSubscriber.kt

```diff
@@ -57,25 +58,24 @@ class RabbitMessageGroupBatchSubscriber(
             return batch
         }
 
-        val groups = batch.groupsList.asSequence()
-            .filter { group ->
-                group.messagesList.all { message ->
-                    callFilterFunction(message, filters)
-                }.also { allMessagesMatch ->
-                    if (!allMessagesMatch) {
-                        logger.debug { "Skipped message group because none or some of its messages didn't match any filters: ${group.toJson()}" }
-                        incrementDroppedMetrics(
-                            group.messagesList,
-                            th2Pin,
-                            MESSAGE_DROPPED_SUBSCRIBE_TOTAL,
-                            MESSAGE_GROUP_DROPPED_SUBSCRIBE_TOTAL
-                        )
-                    }
+        val newBatch = MessageGroupBatch.newBuilder()
+
+        batch.groupsList.forEach { group ->
+            val dropped = group.filterMessagesToNewGroup(newBatch) { callFilterFunction(it, filters) }
+            if (dropped.isNotEmpty()) {
+                logger.debug {
+                    "Skipped ${dropped.size} message(s) from group because it(they) didn't match any filters: ${dropped.joinToString { it.toJson() }}"
```

Comment:
Message can be too large. Please print short version of it, for example metadata or message id.
NOTE: As I know users set unique sequence to message id in scripts to trace messages

Reply:
Changed to `logId`